grouper/audit.py

```diff
@@ -185,3 +201,80 @@ def get_auditors_group(settings, session):
     if not any([p.name == PERMISSION_AUDITOR for p in group.my_permissions()]):
         raise GroupDoesNotHaveAuditPermission()
     return group
+
+
+def get_group_audit_members_infos(session, group):
+    # type: (Session, Group) -> List[AuditMemberInfo]
+    """Get audit information about the members of a group.
+
+    Note that only current members of the group are relevant, i.e., members of the group at the
+    time the current audit was started but are no longer part of the group are excluded, as are
+    members of the group added after the audit was started.
+
+    Arg(s):
+        session: The SQL session
+        group: The group
+
+    Return:
+        List of AuditMemberInfo.
+
+    """
+    members_edge_ids = {member.edge_id for member in itervalues(group.my_members())}
+    user_members = (
+        session.query(AuditMember, GroupEdge._role, User)
+        .filter(
+            AuditMember.audit_id == group.audit_id,
+            AuditMember.edge_id == GroupEdge.id,
+            GroupEdge.member_type == OBJ_TYPES["User"],
+            GroupEdge.member_pk == User.id,
+            # but only those members who are still current members of the group, because members
+            # that need audits might have left the group after the audit was started
+            AuditMember.edge_id.in_(members_edge_ids),
+        )
+        .all()
+    )
+
+    group_members = (
+        session.query(AuditMember, GroupEdge._role, Group)
+        .filter(
+            AuditMember.audit_id == group.audit_id,
+            AuditMember.edge_id == GroupEdge.id,
+            GroupEdge.member_type == OBJ_TYPES["Group"],
+            GroupEdge.member_pk == Group.id,
+            # but only those members who are still current members of the group, because members
+            # that need audits might have left the group after the audit was started
+            AuditMember.edge_id.in_(members_edge_ids),
+        )
+        .all()
+    )
+
+    return [
+        AuditMemberInfo(audit_member, audit_member_role, member_obj)
+        for audit_member, audit_member_role, member_obj in itertools.chain(
+            user_members, group_members
+        )
+    ]
+
+
+def get_group_audit_members_count(session, group, status):
```

Comment:
How about making this something like `group_has_pending_audit_members`? Since there's only the single callsite, making it generic across statuses and returning the full count seems a bit unnecessary

Reply:
sgtm